ppr-ui/src/views/newMhrRegistration/SubmittingParty.vue

```diff
@@ -96,18 +132,53 @@ export default defineComponent({
 
     const localState = reactive({
       attentionReferenceNum: '',
+      documentId: '',
+      isDocumentIdValid: false,
       isRefNumValid: false,
+      documentIdRules: computed(() => customRules(required('Enter a Document ID'), maxLength(8, true), isNumber())),
       validateSubmitter: computed(() => {
         return getSectionValidation(MhrSectVal.SUBMITTING_PARTY_VALID, MhrCompVal.SUBMITTER_VALID)
       }),
+      validateDocId: computed(() => {
+        return getSectionValidation(MhrSectVal.SUBMITTING_PARTY_VALID, MhrCompVal.DOC_ID_VALID)
+      }),
       validateRefNum: computed(() => {
         return getSectionValidation(MhrSectVal.SUBMITTING_PARTY_VALID, MhrCompVal.REF_NUM_VALID)
       })
     })
 
-    watch(() => localState.attentionReferenceNum, (val: string) => {
-      setMhrAttentionReferenceNum(val)
-    })
+    watch(
+      () => localState.documentId,
+      (val: string) => {
+        if (localState.documentId.length === 8) {
+          console.log('Looking up Document ID...')
+          // TODO: Implement Document ID look up
+        }
+        setMhrRegistrationDocumentId(val)
+      }
+    )
+
+    watch(
+      () => localState.isDocumentIdValid,
+      (val: boolean) => {
+        setValidation(MhrSectVal.SUBMITTING_PARTY_VALID, MhrCompVal.DOC_ID_VALID, val)
+      }
+    )
+
+    watch(
+      () => localState.validateDocId,
+      async () => {
+        // @ts-ignore - function exists
+        await context.refs.documentIdForm.validate()
+      }
+    )
+
+    watch(
+      () => localState.attentionReferenceNum,
+      (val: string) => {
+        setMhrAttentionReferenceNum(val)
+      }
+    )
```

Comment:
please revert the format here

Reply:
Manually updated and saved without formatting. It might cause formatting discrepancies again in the future. I'm using prettier settings `/ppr/ppr-ui/prettier.config.js`. Which formatter/setting you are using? I tried multiple and none could format the way yours does...

Reply:
I don't use any.  
Webstorm has it's own configurations that help with imports and whatnot, as well as following the defined linting rules but otherwise i'm not a fan of relying on something to format the code.
Prettier is opinionated code formatting but i'm not sure i agree with the configuration. I'm all for it if it helps with readability/clean code but in this case, my opinion is that it isn't helping with readability.

Reply:
We should consolidate and use the same formatter/settings, so we don't have these issues. Otherwise, everyone on the team will have formatting changes and they commit the code.

Reply:
For sure, although the only discrepancies to date has been with prettier. It's not something we've relied on in other projects but if you feel strongly about this, we can get all get onboard with it. 

To do that though, we will need to adjust and agree on the format output by prettier, as of yet i'm not convinced on its formatting.

Reply:
To add to this, doesn't the linting already provide enough to enable consistency between devs?

Reply:
The `ppr-ui` comes with Prettier and a config `/ppr/ppr-ui/prettier.config.js` when cloned. So during the initial setup, I figured we are using these to format the code across the team. I don't mind the format/style or tool, as long we are consistent across. Let's have a chat sometime.